metpy/plots/skewt.py

```diff
@@ -770,10 +771,70 @@ def plot(self, u, v, **kwargs):
         u, v = delete_masked_points(u, v)
         return self.ax.plot(u, v, **line_args)
 
-    def plot_colormapped(self, u, v, c, **kwargs):
+    def plot_layers(self, u, v, hgt, bounds, colors, **kwargs):
+        r"""Plot different layers in different colors.
+
+        Plots the wind data on the hodograph colored
+        differently for different height layers. Designed to
+        accept layer bounds in meters AGL, to correspond with
+        storm relative helicity calculations. Interpolates to find
+        the points on the hodograph which correspond exactly with the
+        requested height bounds.
+
+        Parameters
+        ----------
+        u : array_like
+            u-component of wind
+        v : array_like
+            v-component of wind
+        hgt: array_like
+            heights from sounding
+        bounds: array_like
+            array of layer bounds, starting
+            at the bottom of the first layer
+            and ending at the top of the last.
+        colors: array_like
+            array of strings containing the
+            colors for each layer of the
+            hodograph
+        kwargs
+            Other keyword arguments to pass to :meth:`matplotlib.axes.Axes.plot`
+
+        Returns
+        -------
+        list[matplotlib.lines.Line2D]
+            lines plotted
+
+        See Also
+        --------
+        :meth:`Hodograph.plot_colormapped`
+
+        """
+        u, v, hgt = delete_masked_points(u, v, hgt)
+        cmap = mpl.colors.ListedColormap(colors)
+        bounds = np.asarray(bounds + hgt[0]) * bounds.units
+        interp_vert = interp(bounds, hgt, hgt, u, v)
+        inds = np.searchsorted(hgt.magnitude, bounds.magnitude)
+        u = np.insert(u.magnitude, inds, interp_vert[1].magnitude)
```

Comment:
Wouldn't it be more efficient to insert the bounds into heights and then interp? That way you'd get all the points plus the new ones for u and v from `interp`.